src/main/java/io/cdap/plugin/servicenow/restapi/RestAPIResponse.java

```diff
@@ -48,19 +50,19 @@ public class RestAPIResponse {
     "    },\n" +
     "    \"status\": \"failure\"\n" +
     "}";
-  private int httpStatus;
-  private Map<String, String> headers;
-  private String responseBody;
-  private boolean isRetryable;
+  private final int httpStatus;
+  private final Map<String, String> headers;
+  private final String responseBody;
+  private final boolean isRetryable;
 
-  public RestAPIResponse(int httpStatus, Map<String, String> headers, String responseBody) {
+  public RestAPIResponse(int httpStatus, Map<String, String> headers, String responseBody) throws IOException {
     this.httpStatus = httpStatus;
     this.headers = headers;
     this.responseBody = responseBody;
-    this.checkRetryable();
+    this.isRetryable = isRetryable(responseBody);
```

Comment:
Shouldn't the `isRetryable` also take `httpStatus` into account?

Reply:
I feel like you should unify the retryable check logic all into this class instead of scattered across this and the RESTApiClient.

Reply:
Refactored all functionality related to APIs to `RestAPIResponse` class.
removed `isRetryable` and `isSuccess` variables/methods. `RestAPIResponse.parse()` function returns `RestAPIResponse` object only when there are no errors and throws `RetryableException` or `IOException` depending on the error being retryable or not.